health/statuscheck/statusidx/agentstatus_idxmap.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2017 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statusidx
+
+import (
+	"github.com/ligato/cn-infra/core"
+	"github.com/ligato/cn-infra/health/statuscheck/model/status"
+	"github.com/ligato/cn-infra/idxmap"
+	"github.com/ligato/cn-infra/idxmap/mem"
+	"github.com/ligato/cn-infra/logging/logroot"
+	log "github.com/ligato/cn-infra/logging/logrus"
+)
+
+// PluginStatusIdxMap provides read-only access to mapping between software interface indexes (used internally in VPP)
```

Comment:
multiple comments in the file refer to sw interface indexes instead of plugin's status